ee/codegen/src/context/workflow-output-context.ts

```diff
@@ -23,6 +30,27 @@ export class WorkflowOutputContext {
   }
 
   public getOutputName(): string {
-    return toSnakeCase(this.terminalNodeData.data.name);
+    const name = this.generateUniqueFinalOutputName();
+    return toSnakeCase(name);
+  }
+
+  private generateUniqueFinalOutputName(): string {
+    const value = this.workflowContext.getOutputName(this.terminalNodeData.id);
+    if (value !== undefined) {
+      return value;
+    } else {
+      let counter = 1;
+      const names = new Set(this.workflowContext.outputNamesById.values());
+
+      const originalName = this.terminalNodeData.data.name;
+      let uniqueName = originalName;
+
+      while (names.has(uniqueName)) {
+        uniqueName = `${originalName}-${counter}`;
+        counter++;
+      }
+      this.workflowContext.addOutputName(this.terminalNodeData.id, uniqueName);
+      return uniqueName;
+    }
```

Comment:
This is where the changes happen. Would love feedback on this approach as this is kind of hacky and doesn't really really hit the elegant approach